### PR TITLE
Update all the Go dependencies

### PR DESCRIPTION
**Pull request by dbussink**
## Description

This seems to work for the basics just fine, so let's have CI take a run at this as well to update these.

Only one small update to the Azure blob storage handling seems needed so far.

## Related Issue(s)

Some more critical changes have been separate updated already in for example #11694 

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by deepthi** [comment]
I think this is a good idea. We can review after the tests are fixed (since they look like they might need actual code changes to pass).

**Comment by dbussink** [comment]
> I think this is a good idea. We can review after the tests are fixed (since they look like they might need actual code changes to pass).

@deepthi Yeah, the main issue is that we have many places that end up using `reflect.DeepEqual` with protobufs which is not supported and only kinda sometimes work. I've updated a lot of cases here but we still have many more of those cases that could break at any change in the protobuf package.

**Comment by dbussink** [comment]
This is now ready for review. The issue in the `vtgate_queries` is unrelated to the changes here, but @harshit-gangal is looking at that separately.

**Review by systay — APPROVED** [review]
Thanks for doing this! ❤️

File: go/sqltypes/result.go [hunk 1]
```diff
@@ -90,10 +90,11 @@ func (result *Result) ReplaceKeyspace(keyspace string) {
...
 func (result *Result) Copy() *Result {
 	out := &Result{
-		InsertID:            result.InsertID,
 		RowsAffected:        result.RowsAffected,
-		Info:                result.Info,
+		InsertID:            result.InsertID,
 		SessionStateChanges: result.SessionStateChanges,
+		StatusFlags:         result.StatusFlags,
+		Info:                result.Info,
 	}
```
Comment (harshit-gangal) [inline go/sqltypes/result.go:98]: looks like unrelated changes for the PR `StatusFlags: result.StatusFlags`

**Review by harshit-gangal — APPROVED** [review]

**Review by notfelineit — APPROVED** [review]

**Review by ajm188 — APPROVED** [review]

**Review by frouioui — APPROVED** [review]
